main/lsp/requests/code_action.cc

```diff
@@ -105,12 +326,33 @@ unique_ptr<ResponseMessage> CodeActionTask::runRequest(LSPTypecheckerInterface &
             auto action = make_unique<CodeAction>("Apply all Sorbet fixes for file");
             action->kind = kind;
             auto workspaceEdit = make_unique<WorkspaceEdit>();
-            workspaceEdit->documentChanges = getEdits(config, gs, allEdits);
+            workspaceEdit->documentChanges = getQuickfixEdits(config, gs, allEdits);
             action->edit = move(workspaceEdit);
             result.emplace_back(move(action));
         }
     }
 
+    if (config.opts.lspExtractMethodEnabled) {
+        auto queryResult = queryByLoc(typechecker, params->textDocument->uri, *params->range->start,
+                                      LSPMethod::TextDocumentCodeAction, false);
+
+        // Generate "Extract method" code actions only for method definitions
+        if (queryResult.error == nullptr) {
+            for (auto &resp : queryResult.responses) {
+                if (auto def = resp->isDefinition()) {
+                    if (def->symbol.isMethod()) {
```

Comment:
We currently only return `DefinitionResponse` for method symbols right now (in DefLocSaver)`.

If we change `DefinitionResponse` to store a `MethodRef` instead of a `SymbolRef`, we can get rid of this `if` check, and ensure that if someone wants to make `DefinitionResponse` do something with non-methods in the future, our code here in code_action.cc fails with type errors instead of runtime logic errors.

Reply:
https://github.com/sorbet/sorbet/pull/5473